x/profile/keeper/msg_server_set_validator_cons_address.go

```diff
@@ -0,0 +1,65 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	valtypes "github.com/tendermint/spn/pkg/types"
+	"github.com/tendermint/spn/x/profile/types"
+)
+
+func (k msgServer) SetValidatorConsAddress(
+	goCtx context.Context,
+	msg *types.MsgSetValidatorConsAddress,
+) (*types.MsgSetValidatorConsAddressResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	valPubKey, err := valtypes.NewValidatorConsPubKey(msg.ValidatorConsPubKey, msg.ValidatorKeyType)
+	if err != nil {
+		return &types.MsgSetValidatorConsAddressResponse{},
+			sdkerrors.Wrap(types.ErrInvalidValidatorKey, string(msg.ValidatorConsPubKey))
```

Comment:
I think it can be a critical error since this check is performed in `Validate` message